File: code_snippets/sprinkler.ml
let sprinkler_model =
  let* cloudy    = bernoulli 0.5 in
  let* rain      = bernoulli (if cloudy then 0.8 else 0.2)
  and* sprinkler = bernoulli (if cloudy then 0.1 else 0.5) in
  let* wet_grass = bernoulli (match rain, sprinkler with
      | true,  true  -> 0.99
      | true,  false -> 0.9
      | false, true  -> 0.9
      | false, false -> 0.0
    )
  in
  condition wet_grass
    (return rain)

File: code_snippets/webppl/sprinkler.js
var sprinkler_model = function (method) {
  return function () {
    var sprinkler = function () {
      var cloudy = bernoulli({ p: 0.5 });
      var rain = cloudy
          ? bernoulli({ p: 0.8 })
          : bernoulli({ p: 0.2 });
      var sprinkler = cloudy
          ? bernoulli({ p: 0.1 })
          : bernoulli({ p: 0.5 });
      var wet_grass = rain
          ? (sprinkler? 0.99 : 0.9)
          : (sprinkler? 0.9  : 0.);
      condition(wet_grass);
      return rain;
    };
    return Infer(method, sprinkler);
  };
};
